be/src/exprs/agg/factory/aggregate_factory.hpp

```diff
@@ -66,410 +66,407 @@ class AggregateFactory {
 public:
     // The function should be placed by alphabetical order
     template <LogicalType LT>
-    static auto MakeAvgAggregateFunction() {
-        return std::make_shared<AvgAggregateFunction<LT>>();
+    static const AggregateFunction* MakeAvgAggregateFunction() {
```

Comment:
Changing `auto` to `const AggregateFunction*` will cause the elimination of virtual function (#9080)  cannot be effective.

Reply:
fixed